test/functional/dashboard_controller_test.rb

```diff
@@ -1,8 +1,21 @@
 require 'test_helper'
 
 class DashboardControllerTest < ActionController::TestCase
-  test "should get index" do
+  test 'should get index' do
     get :index, {}, set_session_user
     assert_response :success
   end
+
+  test 'create returns 404 if widget to add is not found' do
+    post :create, { :name => 'non-existent-widget' }, set_session_user
+    assert_response :not_found
+  end
+
+  test 'create adds widget to user if widget is valid' do
+    User.current = users(:admin)
```

Comment:
Not sure this is needed?  Or if it is for something (I can't see what), then use `as_user` or `as_admin`.